agent/acl_endpoint_test.go

```diff
@@ -1222,6 +1222,26 @@ func TestACL_LoginProcedure_HTTP(t *testing.T) {
 			methodMap[method.Name] = method
 		})
 
+		t.Run("Create in invalid datacenter", func(t *testing.T) {
```

Comment:
Not a blocker, but I believe the purpose of this test is to show we accept a `dc` query parameter, right? Since we don't have a backend interface we can fake, and we don't want to have to setup a whole second server in a different DC the test happens to fail with invalid DC. But I believe the underlying thing we are testing is that `parseDC` is called and sets the DC value correct. The current name suggests it is more about testing validation of the `dc` query parameter, but I think that might be more incidental than the primary purpose of the case.

I would suggest changing the name to something like this:

```suggestion
		t.Run("Create in remote DC", func(t *testing.T) {
```

Reply:
That is a good catch. I was testing incidental behavior purely for the reasons you have described. Ill rename these tests to reflect that behavior they do test.

Reply:
i might have spoken too soon! the refactor seems large, and ill punt on it for now. Thanks so much for the heads up!